src/app/recents/recents.component.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { fakeAsync, ComponentFixture, TestBed, tick } from '@angular/core/testing';
+import { DatePipe } from '@angular/common';
+import { TabsModule } from 'ngx-bootstrap/tabs';
+import { of } from 'rxjs';
+import { NavigationEnd } from '@angular/router';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { BsDropdownModule } from 'ngx-bootstrap/dropdown';
+
+import { RecentsComponent } from './recents.component';
+import { RecentsService } from '../services/recents.service';
+import { recents } from './test.data';
+import { NodeTypeDisplayPipe, NodeExternalUrlPipe, TruncatePipe } from '../pipes/nodedisplay';
+import { ArtifactsTableComponent } from '../tables/artifacts-table/artifacts-table.component';
+import { EstuaryTableComponent } from '../tables/table.component';
+import { PropertyDisplayPipe, PropertyValueDisplayPipe } from '../pipes/propertydisplay';
+
+
+fdescribe('RecentsComponent', () => {
+  let fixture: ComponentFixture<RecentsComponent>;
+  let recentsService: RecentsService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ArtifactsTableComponent,
+        EstuaryTableComponent,
+        NodeTypeDisplayPipe,
+        NodeExternalUrlPipe,
+        PropertyDisplayPipe,
+        PropertyValueDisplayPipe,
+        RecentsComponent,
+        TruncatePipe
+      ],
+      providers: [
+        RecentsService,
+        DatePipe
+      ],
+      imports: [
+        TabsModule.forRoot(),
+        HttpClientTestingModule,
+        NoopAnimationsModule,
+        BsDropdownModule.forRoot()
+      ],
+      // This is used to avoid having to import components we don't test here
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+    recentsService = TestBed.get(RecentsService);
+  });
+
+  it('should retrieve the 5 most recent advisories', fakeAsync(() => {
+    spyOn(recentsService, 'getRecents').and.returnValue(
+      of(recents)
+    );
+    fixture = TestBed.createComponent(RecentsComponent);
+    fixture.detectChanges();
+    const component = fixture.componentInstance;
+    expect(component.recentsAll).toBe(<any> recents.data);
```

Comment:
What is the `<any>` for?

Reply:
@mprahl Honestly, I'm not sure :sweat_smile: But it doesn't work without it because it gives me a type mismatch.

Reply:
Fair enough haha

Reply:
@sarah256 ah, I think it's because in `recents.component.ts`, you declared the type of `recentsAll` as `Array<any>`, but it's actually an object. So you can just use `any` as the type when declaring `recentsAll`. It looks like the type for `meta` needs to be adjusted as well.